tests/integration/p2p.go

```diff
@@ -398,12 +413,16 @@ func setupReplicatorWaitSync(
 		ready <- struct{}{}
 		for waitIndex := 0; waitIndex < len(sourceToTargetEvents); waitIndex++ {
 			for i := 0; i < targetToSourceEvents[waitIndex]; i++ {
-				err := sourceNode.WaitForPushLogByPeerEvent(targetPeerInfo.ID)
-				require.NoError(s.t, err)
+				msg, ok := <-sourceSub.Message()
+				if ok {
+					assert.Equal(s.t, targetPeerInfo.ID, msg.Data.(events.MergeEvent).ByPeer)
```

Comment:
suggestion: I think `require` would be more appropriate here (and the other similar blocks in this file) - a failure here is a bug in the test, not a bug in the production code?  It would also probably be really confusing if the code continued past this point no?

todo: Please document why this assert/require is done